components/jenkins/keyvault.tf

```diff
@@ -47,3 +47,10 @@ resource "azurerm_key_vault_secret" "subscription_id" {
   value        = data.azurerm_client_config.current.subscription_id
   key_vault_id = azurerm_key_vault.jenkinskv.id
 }
+
+resource "azurerm_key_vault_secret" "env_subscription_id" {
+  count        = var.env == "ptl" ? 1 : 0
```

Comment:
this should be done in the other component for all environments just prefixing it with "${var.env}-subscription-id"